components/x-gift-article/src/lib/highlightsApi.js

```diff
@@ -54,4 +56,20 @@ export default class HighlightsApiClient {
 			return { annotationsCopyResult: undefined }
 		}
 	}
+
+	async shareHighlights(articleId) {
+		try {
+			const response = await this.fetchJson('/create-token', {
+				method: 'POST',
+				body: JSON.stringify({ articleId })
+			})
+
+			if (response) {
+				return response
+			}
+			return {}
```

Comment:
**Question:** Given how the `fetchJson` function behaves is there any scenario where `response` would not be defined other than in a error scenario? https://github.com/Financial-Times/x-dash/blob/05680b0268774dbf50a4bfeb233390f7c44a5a47/components/x-gift-article/src/lib/highlightsApi.js#L28-L43

Reply:
Hmm, I don't think so , since if the `response` is undefined it will through when calling  `response.json()` , will remove the return statement and the if condition then